### ISSUE TITLE
Go parser cannot parse go.mod and go.sum files without a namespace

### ISSUE THREAD

**Issue by jmastr** [nexB/scancode-toolkit#2229]
### Description

> Please leave a brief description of the bug or feature request:

Normally an entry to e.g. `go.sum` looks something like this:
```
github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
```
The Go parser parses `github.com/BurntSushi` into `namespace` and `toml` into `name`.

I have this entry:
```
go.opencensus.io v0.21.0/go.mod h1:mSImk1erAIZhrmZN+AvHh14ztQfjbGwt4TtuofqLduU=
```
Basically the `namespace` portion is missing.

For my concrete issue this is the corresponding source code: https://github.com/nexB/scancode-toolkit/blob/v3.2.1rc2/src/packagedcode/go_mod.py#L253-L254, but `namespace` is used through out the Go parser.

My proposal is to remove the `namespace` portion and parse everything into `name`, since I do not see in the source that `namespace` is used standalone somewhere.

### How To Reproduce

> Tell us how to reproduce the issue. 

### System configuration

> For bug reports, it really helps us to know:

* What OS are you running on? (Windows/MacOS/Linux) Ubuntu 20.04
* What version of scancode-toolkit was used to generate the scan file? v3.2.1rc2
* What installation method was used to install/run scancode? (pip/source download/other) `./configure`

**Comment by jmastr** [comment]
@rpotter12 Could you have a look at my proposal, please? As far as I can see you are the Go parser maintainer.

**Comment by rpotter12** [comment]
@jmastr Good catch:)
But I think it's better that we should store everything by the way they should be store i.e. namespace should be in namespace and name should be in name. We are not using namespace standalone for now but it will be easier for us in the future if namespace is  needed somewhere else. I think we should change the regular expression in such a way that it detects everything perfectly. :)

**Comment by jmastr** [comment]
@rpotter12 Thanks. 

Could you change the regexes for go.mod and go.sum, so that the namespace can be empty then?

**Comment by rpotter12** [comment]
@jmastr Yes, I will do it soon. :)

**Comment by rpotter12** [comment]
@jmastr Can you give a link of file where `this go.opencensus.io v0.21.0/go.mod h1:mSImk1erAIZhrmZN+AvHh14ztQfjbGwt4TtuofqLduU=` is used?

**Comment by pombredanne** [comment]
@rpotter12 here are links :
- https://github.com/census-instrumentation/opencensus-service/blob/57e037c599b75fe2dddf89cb9ada67bf4c8de9fc/go.mod
- https://github.com/census-instrumentation/opencensus-service/blob/57e037c599b75fe2dddf89cb9ada67bf4c8de9fc/go.sum

**Comment by rpotter12** [comment]
In https://github.com/nexB/scancode-toolkit/pull/2230/files#diff-081f69678b5dd083e0c798c6e4d24645R126 we should use a+ but it doesn't work for me.

I think it will work for all the cases because r'(?P[^\s]+)(\s)+(?P<ns_name>[^\s]+)\s?(?P(.))' in this \s? is used because module line is not having version number so version will be optional in that case and require and exclude line (?P(.)) will detect all the characters till end.

**Comment by pombredanne** [comment]
@jmastr thank you for the report and @rpotter12 thank you for the fix with #2230
@jmastr would you be kind enough to test if the latest develop branch works for you?

**Comment by jmastr** [comment]
@pombredanne just checked. it works for me. @rpotter12 thank you for the solution!
